nova/scheduler/filters/cpu_info_migration_filter.py

```diff
@@ -50,17 +51,26 @@ def filter_all(self, filter_obj_list, spec_obj):
             return filter_obj_list
 
         try:
-            ctx = get_admin_context()
-            compute_node = ComputeNode.get_by_host_and_nodename(
-                ctx, source_host, source_node)
-            source_cpu_flags = self._parse_cpu_info(compute_node.cpu_info)
-        except ComputeHostNotFound:
+            context = nova_context.get_admin_context()
+            host_mapping = HostMapping.get_by_host(context, source_host)
+            cell_mapping = host_mapping.cell_mapping
+            with nova_context.target_cell(context, cell_mapping) as cctxt:
+                compute_node = ComputeNode.get_by_host_and_nodename(
+                    cctxt, source_host, source_node)
+                try:
+                    source_cpu_flags = self._parse_cpu_info(
+                        compute_node.cpu_info)
+                except (ValueError, KeyError):
+                    LOG.warning(
+                        "Cannot parse cpu_info for source host/node %s/%s: %s",
+                        source_host, source_node, compute_node.cpu_info)
+                    return []
```

Comment:
I'd move this out of the current try-except block, because 1) it shouldn't need the cell-context anymore and 2) the inner and outer try-except look distinct i.e. the outer wouldn't catch anything done in this inner one and 3) to keep the outer try-except as short as needed

Reply:
I moved it explicitly inside, because I only want to catch those exception raised by `_parse_cpu_info`, and not any other unrelated ones, such as anything done by entering the cell-context.
That happened during the unit-tests, where it was trying to log something, which raised a `KeyError`-exception, which was handled by the outer exception block. From my point-of-view, that is incorrect behaviour

Reply:
Ah, okay. Sure, it's done